site/frontend/src/pages/dashboard.ts

```diff
@@ -3,6 +3,22 @@ import {DASHBOARD_DATA_URL} from "../urls";
 
 import {getJson} from "../utils/requests";
 
+type ScaleKind = "linear" | "log";
+let scale: ScaleKind = "linear";
+
+const buttons = Array.from(
+  document.querySelectorAll<HTMLInputElement>("#scale-select-form input")
+);
+
+buttons.map((button) => {
+  button.addEventListener("change", () => {
+    if (button.checked) {
+      scale = button.value as ScaleKind;
+      make_data();
```

Comment:
Could you please refactor the code so that we only re-render the data when changing the axes, rather than downloading it again from the server? It's a bit wasteful :) We can store the response into a variable, and then just call `populate_data` on it again.

Reply:
Thanks, I've added `response` var to cache the response because of the `getJson` fn is async.